pkg/kube/cluster-init.sh

```diff
@@ -83,6 +120,14 @@ setup_prereqs
 
 date >> $INSTALL_LOG
 HOSTNAME=$(/bin/hostname)
+logmsg "Starting wait for hostname, currently: $HOSTNAME"
+while [[ $HOSTNAME = linuxkit* ]];
```

Comment:
What are you trying to check?
It is the case that EVE comes up with a linuxkit* name early in boot, but then the hostname is changed to be the device UUID string. So I wouldn't depend on it being linuxkit* if you are trying to check whether this is running EVE-OS.

Reply:
If you want to check whether or not the device has been onboarded then we have a way to do that using pubsub and you can potentially use the pubsub checkpoint file in /persist/status/zedclient/OnboardingStatus/global.json to check this.

Reply:
I could have added a good comment in the script there.  This was added to avoid a hang further down in startup where k3s node ready state was polled looking for a node in the ready state with the linuxkit* name (which doesn't exist anymore at this point in runtime).  The k3s check ready path has been refactored since the hostname poll change was put in place.  With fresh eyes now I think a smaller change could be in check_node_ready() and just read a fresh copy of the system hostname.